src/humanloop/otel/processor.py

```diff
@@ -0,0 +1,194 @@
+import json
+import logging
+from collections import defaultdict
+from typing import Any
+
+# No typing stubs for parse
+import parse  # type: ignore
+from opentelemetry.sdk.trace import ReadableSpan
+from opentelemetry.sdk.trace.export import SimpleSpanProcessor, SpanExporter
+
+from humanloop.otel.constants import HL_FILE_OT_KEY, HL_LOG_OT_KEY
+from humanloop.otel.helpers import (
+    is_humanloop_span,
+    is_llm_provider_call,
+    read_from_opentelemetry_span,
+    write_to_opentelemetry_span,
+)
+
+
+class HumanloopSpanProcessor(SimpleSpanProcessor):
+    """Enrich Humanloop spans with data from their children spans.
```

Comment:
Could you explain further what you mean by this?

Why are we enriching parents with data from their children?

Reply:
ack, expanded the docstring